app/src/main/java/org/mozilla/focus/experiments/ExperimentalFeatures.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.experiments
+
+import mozilla.components.service.nimbus.NimbusApi
+
+private const val EXPERIMENT_SINGLE_TAB_BROWSER = "single-tab-browser"
+private const val BRANCH_TREATMENT = "treatment"
+
+/**
+ * Feature flags driven by Nimbus experiments.
+ */
+class ExperimentalFeatures(nimbus: NimbusApi) {
+    /**
+     * Whether or not this browser should have multiple tabs or run in single-tab mode.
+     */
+    val hasMultipleTabs: Boolean by lazy {
+        // We are wrapping this in "by lazy" because we do not want to switch between a
+        // multi-tab/single-tab browser at runtime while the user is using the browser. Instead we
+        // decide once on startup.
+
+        @Suppress("TooGenericExceptionCaught")
+        val branch = try {
+            nimbus.getExperimentBranch(EXPERIMENT_SINGLE_TAB_BROWSER)
```

Comment:
I would strongly suggest that you don't use this since it ties the application to experiment internals. We have deprectated the use of the `getExperimentBranch` function in favor of using the Feature API: https://experimenter.info/feature-variables-and-me

Reply:
Here's an example of the Feature API instrumentation in Fenix for a reference: https://github.com/mozilla-mobile/fenix/blob/3f9d277566cceed8449c63b19939afba4c8ea104/app/src/main/java/org/mozilla/fenix/experiments/NimbusFeatures.kt